src/main/java/xyz/kmahyyg/eshopdemo/controller/OrderController.java

```diff
@@ -1,29 +1,102 @@
 
-import java.util.List;
+import javax.servlet.http.HttpServletResponse;
+import java.math.BigDecimal;
+import java.util.*;
 
 @Controller
 public class OrderController {
     @Autowired
     private SysOrdersDao sysOrdersDao;
 
     @Autowired
-    private UserInfoUtil userInfoUtil;
-
-//    @RequestMapping("/show/user/order")
-//    public String showOrderOfCurrentUser() {
-//            String currentUserUid = userInfoUtil.getCurrentUserID();
-//            if (!currentUserUid.isEmpty()){
-//                List<SysOrders> allOrdersByUser = sysOrdersDao.selectByUserId(currentUserUid);
-//            }
-//            //TODO: render the orders data in template html
-//        return "userorders";
-//    }
+    private SysUsersDao sysUsersDao;
+
+
+    @Autowired
+    private SysItemsDao sysItemsDao;
+
+
+    /**
+     * 查看当前用户的历史订单
+     * @param model
+     * @return
+     */
+    @GetMapping("/show/user/order")
+    public String showOrderOfCurrentUser(Model model) {
+        UserInfo currentUser = (UserInfo)SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (currentUser != null) {
+            String currentUsername = currentUser.getUsername();
+            if (!currentUsername.isEmpty()){
+                String currentUserUid = sysUsersDao.selectByUserName(currentUsername).getUid();
+                List<SysOrders> allOrdersByUser = sysOrdersDao.selectByUserId(currentUserUid);
+                model.addAttribute("orders",allOrdersByUser);
+            }
+        }
+        return "userorders";
+    }
+
+
+    @RestController
+    @RequestMapping("/api/user")
+    public class orderController {
+        @PostMapping("/order")
+        public String addOrder(@RequestBody String jsonParam) throws Exception {
+            UserInfo currentUser = (UserInfo) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+            if (currentUser != null) {
+                String currentUsername = currentUser.getUsername();
+                if (!currentUsername.isEmpty()) {
+                    String currentUserUid = sysUsersDao.selectByUserName(currentUsername).getUid();
+                    SysOrders record = new SysOrders();
+                    record.setOid(UUID.randomUUID().toString());
+                    record.setStatus(OrderStatusEnum.CREATED);
+                    record.setItems(jsonParam);
+                    record.setUid(currentUserUid);
+                    /**
+                     * 从购物车中拿到商品id和数量，计算价格
+                     */
+                    JSONObject jsonObject = new JSONObject(jsonParam);
+                    JSONArray cart = jsonObject.getJSONArray("items");
+                    for (int i = 0; i < cart.length(); i++) {
+                        JSONObject jsonobj = (JSONObject) cart.get(i);
+                        Integer itemid = Integer.valueOf(jsonobj.getString("itemId"));
+                        Integer itemno = Integer.valueOf(jsonobj.getString("itemNo"));
```

Comment:
Check about the previous updated commit in the upstream. I've already made some changes to make sure you get an object instead of string here.